numba/tests/test_remove_dead.py

```diff
@@ -218,6 +218,18 @@ def func(A, i):
             # recover global state
             ir_utils.alias_func_extensions = old_ext_handlers
 
+    def test_rm_dead_rhs_vars(self):
+        """make sure lhs variable of assignment is considered live if used in
+        rhs (test for #6715).
+        """
+        @numba.njit
+        def func():
+            for i in range(3):
+                a = (lambda j: j)(i)
+                a = np.array(a)
+
+        func()
```

Comment:
Perhaps return `a` and assert its value? Else this just confirms that it compiles opposed to it having compiled correctly?

Reply:
Done.

Reply:
thanks